modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java

```diff
@@ -14251,6 +14516,78 @@ private void _testGetObjectEntryWithObjectActions(
 		unsafeTriConsumer.accept(actionJSONObject, jsonObject, objectAction);
 	}
 
+	private void _testPatchPutScopeScopeKeyByExternalReferenceCode(
+			Http.Method httpMethod)
+		throws Exception {
+
+		_siteScopedObjectEntry1 = ObjectEntryTestUtil.addObjectEntry(
+			_siteScopedObjectDefinition1, _OBJECT_FIELD_NAME_1,
+			_OBJECT_FIELD_VALUE_1);
+		Group group = _groupLocalService.fetchGroup(
+			TestPropsValues.getGroupId());
+
+		// Using groupExternalReferenceCode
+
+		String newObjectFieldValue = RandomTestUtil.randomString();
+
+		Assert.assertEquals(
+			HTTPTestUtil.invokeToJSONObject(
+				JSONUtil.put(
+					_OBJECT_FIELD_NAME_1, newObjectFieldValue
+				).toString(),
+				StringBundler.concat(
+					_getEndpoint(
+						group.getExternalReferenceCode(), 0,
+						_siteScopedObjectDefinition1),
+					"/by-external-reference-code/",
+					_siteScopedObjectEntry1.getExternalReferenceCode()),
+				httpMethod
+			).getString(
+				_OBJECT_FIELD_NAME_1
+			),
```

Comment:
We should avoid chaining method calls. Have you seen any examples of this pattern in our existing tests?